cdap-api/src/main/java/co/cask/cdap/api/retry/RetryPolicy.java

```diff
@@ -0,0 +1,218 @@
+      ", initialDelay=" + baseDelay +
+      ", maxDelay=" + maxDelay +
+      ", timeout=" + timeout +
+      '}';
+  }
+
+  /**
+   * @return a builder to create a retry policy using the configured CDAP default backoff type.
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * Builds a RetryPolicy.
+   */
+  public static class Builder {
+    protected Type type;
+    protected Long baseDelay;
+    protected Long maxDelay;
+    protected Long timeout;
+
+    /**
+     * Set the type of retry policy. If none is given, the configured CDAP default will be used at runtime.
+     *
+     * @param type the type of retry policy
+     * @return this builder
+     */
+    public Builder setType(Type type) {
+      return this;
+    }
+
+    /**
+     * Set the initial delay between retries. If none is given, the configured CDAP default will be used at runtime.
+     *
+     * @param duration initial delay between retries
+     * @param timeUnit time unit of the duration. Must be in milliseconds or greater
+     * @return this builder
+     */
+    public Builder setBaseDelay(long duration, TimeUnit timeUnit) {
+      validateDuration("baseDelay", duration, timeUnit);
+      baseDelay = TimeUnit.MILLISECONDS.convert(duration, timeUnit);
+      return this;
+    }
+
+    /**
+     * Set the maximum delay between retries.
+     * If none is given, the configured CDAP default will be used at runtime. Has no effect if the backoff
+     * is {@link Type#FIXED_DELAY}.
+     *
+     * @param duration maximum delay between retries
+     * @param timeUnit time unit of the duration. Must be in milliseconds or greater
+     * @return this builder
+     */
+    public Builder setMaxDelay(long duration, TimeUnit timeUnit) {
+      validateDuration("maxDelay", duration, timeUnit);
+      maxDelay = TimeUnit.MILLISECONDS.convert(duration, timeUnit);
+      return this;
+    }
+
+    /**
+     * Set the amount of time to retry before failing.
+     * If none is given, the configured CDAP default will be used at runtime.
+     *
+     * @param duration amount of time to retry before failing
+     * @param timeUnit time unit of the duration. Must be in milliseconds or greater
+     * @return this builder
+     */
+    public Builder setTimeout(long duration, TimeUnit timeUnit) {
+      validateDuration("timeout", duration, timeUnit);
+      timeout = TimeUnit.MILLISECONDS.convert(duration, timeUnit);
+      return this;
+    }
+
+    protected void validateDuration(String property, long duration, TimeUnit timeUnit) {
+      if (duration <= 0) {
+        throw new IllegalArgumentException(
+          String.format("Invalid %s of %d. %s must be a positive number.", property, duration, property));
+      }
+      if (timeUnit == TimeUnit.MICROSECONDS || timeUnit == TimeUnit.NANOSECONDS) {
+        throw new IllegalArgumentException(
+          String.format("Invalid time unit for %s of %s. Time units must be milliseconds or greater",
```

Comment:
was thinking that the corresponding preference would be in millis, but we could just have a timeunit preference as well. will remove.